pkcs11/src/new/types/slot_token.rs

```diff
@@ -14,15 +14,17 @@ impl Slot {
     }
 
     /// Underlying ID used for a slot
-    pub fn id(&self) -> u64 {
+    pub fn id(&self) -> CK_SLOT_ID {
```

Comment:
I agree it would be the best thing to do but we try not to expose any raw bindgen type from the `pkcs11` crate, but to abstract them all.

I'm thinking, if the specs say that the value is at least 32 bits long, do you think we can assume that in most cases the `CK_SLOT_ID` value would be less that `UINT_32_MAX`?

Maybe here we can return `u32` and do:
```rust
self.slot_id.try_from().expect("CK_SLOT_ID value can not fit in 32 bits")
```
(or make this function return `Result<u32>`, we already have conversion between `TryFrom` error and our `Error` type)

and maybe replace `from_u32` with:
```rust
impl From<u32> for Slot {
    fn from(val: u32) -> Self {
        Slot {
            slot_id: val.try_from().expect("CK_SLOT_ID should at least be 32 bits long"),
        }
    }
}
```

Reply:
While it's something I particularly like, we could have two implementations of this type, that get selected at build time? The problem is that, obviously, everything that uses it must also deal with that duality :)

Reply:
@hug-dev Your approach seems even better, I'm happy to incorporate it into the PR.

Reply:
After thinking about it a bit more, I'm not sure whether it's safe to assume that the upper bits are not used. I converted the public interface to `u64` in case someone uses them. The conversion to the raw type is implemented as `From<...>` like the other wrappers.

Reply:
I think the safest to do anyway is to return an error, the overhead is not too big with the auto-converison of error types. Because I think nothing prevents C header to defined `unsigned long` as a 128 bits sometimes in the future...
Good as it is for now!!